### PR TITLE
[Scenes] Default Scene Handler Config

### PR DESCRIPTION
**Pull request by lpbeliveau-silabs**
Added a config allowing to choose if we want to enable the default scenes handler, it is enabled by default.
If a user wants to enable it's own scenes handlers instead of using the default ones, the config must first be set to 0.

### CODE REVIEW

File: src/lib/core/CHIPConfig.h [hunk 1]
```diff
@@ -1457,6 +1457,17 @@ extern const char CHIP_NON_PRODUCTION_MARKER[];
 #endif // CHIP_CONFIG_TEST
 #endif // CHIP_CONFIG_MAX_SCENES_TABLE_SIZE
 
+/**
+ * @def CHIP_CONFIG_MAX_SCENES
+ *
+ * @brief This defines enables the caude for the default scene handlers in each sceneable cluster.
```
Comment (andy31415) [inline src/lib/core/CHIPConfig.h:1463]: s/caude/clause/? 

or something else? I am not sure I can parse this sentence.

**Review by andy31415 — APPROVED** [review]

File: src/lib/core/CHIPConfig.h (hunk 1 above)
Comment (lpbeliveau-silabs) [reply src/lib/core/CHIPConfig.h]: Reworded

File: src/lib/core/CHIPConfig.h [hunk 2]
```diff
@@ -1457,6 +1457,18 @@ extern const char CHIP_NON_PRODUCTION_MARKER[];
 #endif // CHIP_CONFIG_TEST
 #endif // CHIP_CONFIG_MAX_SCENES_TABLE_SIZE
 
+/**
+ * @def CHIP_CONFIG_MAX_SCENES
+ *
+ * @brief This defines enables the automatic registration of default scene handlers in the scene table for each sceneable clusters.
+ * If a user want's to use their own scene handlers, they can disable this flag and implement their own handlers.
```
Comment (mkardous-silabs) [inline src/lib/core/CHIPConfig.h:1464]:
```suggestion
 * If a user wants to use their own scene handlers, they can disable this flag and implement their own handlers.
```

File: src/lib/core/CHIPConfig.h [hunk 3]
```diff
@@ -1457,6 +1457,18 @@ extern const char CHIP_NON_PRODUCTION_MARKER[];
 #endif // CHIP_CONFIG_TEST
 #endif // CHIP_CONFIG_MAX_SCENES_TABLE_SIZE
 
+/**
+ * @def CHIP_CONFIG_MAX_SCENES
+ *
+ * @brief This defines enables the automatic registration of default scene handlers in the scene table for each sceneable clusters.
```
Comment (mkardous-silabs) [inline src/lib/core/CHIPConfig.h:1463]:
```suggestion
 * @brief This defines enables the automatic registration of the default scene handlers in the scene table for each sceneable cluster.
```

File: src/lib/core/CHIPConfig.h [hunk 4]
```diff
@@ -1457,6 +1457,18 @@ extern const char CHIP_NON_PRODUCTION_MARKER[];
...
+ * If a user want's to use their own scene handlers, they can disable this flag and implement their own handlers.
+ * Then they can use ScenesServer::Instance().RegisterSceneHandler() to have their handlers called when a scene is recalled or
```
Comment (mkardous-silabs) [inline src/lib/core/CHIPConfig.h:1465]:
```suggestion
 * They can use ScenesServer::Instance().RegisterSceneHandler() to have their handlers called when a scene is recalled or
```

File: src/lib/core/CHIPConfig.h [hunk 5]
```diff
@@ -1457,6 +1457,18 @@ extern const char CHIP_NON_PRODUCTION_MARKER[];
 #endif // CHIP_CONFIG_TEST
 #endif // CHIP_CONFIG_MAX_SCENES_TABLE_SIZE
 
+/**
+ * @def CHIP_CONFIG_MAX_SCENES
```
Comment (mkardous-silabs) [inline src/lib/core/CHIPConfig.h:1461]:
```suggestion
 * @def CHIP_CONFIG_SCENES_USE_DEFAULT_HANDLERS
```

**Review by mkardous-silabs — COMMENTED** [review]
- [x] all your `#endif // ...` don't match the initial #if ... now

**Review by jmartinez-silabs — APPROVED** [review]

**Review by bzbarsky-apple — APPROVED** [review]
Looks good, with @mkardous-silabs's issues addressed.

**Review by mkardous-silabs — APPROVED** [review]

**Pull request merged**